neo/core/analogsignal.py

```diff
@@ -447,33 +447,38 @@ def _pp(line):
     def time_index(self, t):
         """Return the array index corresponding to the time `t`"""
         t = t.rescale(self.sampling_period.units)
```

Comment:
does t need to be rescaled here? and if so then the invers of the sampling_rate should be used, because this is what this is compared to later.
I think it would be best not to rescale here but do a `rescale('dimensionless') in the line below.

Reply:
good idea. I've pushed a new commit